### PR TITLE
Adding support for glob matching of policies

### PR DESCRIPTION
**Pull request by sdwr98**
Hi all,

We have a large number of Chronos jobs that all need Vault tokens.  Specifying each name individually is unsupportable, so I've added functionality to support glob patterns in policy names.  I've updated the README to show an example.

### CODE REVIEW

File: policy.go [hunk 1]
```diff
@@ -36,9 +52,30 @@ var defaultPolicies = map[string]*policy{
...
 	if pol, ok := p[key]; ok {
 		return pol
-	} else if pol, ok := p["*"]; ok {
+	}
+
+	// Now we're going to check for globs
+	// Order the keys in descending order of length
+	// so that "foobar*" takes precedence over "foo*"
+	// Now organize the keys by length
+	policyKeys := make(policyKeyList, 0, )
```
Comment (nemothekid) [inline policy.go]: Does this line compile? I'm not familiar with this syntax.

**Comment by nemothekid** [comment]
I had a single comment - an initial review of the code looks good, but it doesn't look like you've tried to compile it.

Could you also make sure to `go fmt` the edited go files?

File: policy.go (hunk 1 above)
Comment (sdwr98) [inline policy.go]: Apparently it does, although I didn't mean to leave that comma in.  I took it out.

**Comment by sdwr98** [comment]
It does compile and work, although I didn't mean to leave that comma in there.  I formatted the file and updated the PR.  Thanks!

**Comment by nemothekid** [comment]
Thanks for your contribution! I really see how this could help, especially because I don't think Chronos uses easy to guess task names.

**Pull request merged**